src/shapes/SurfaceShape.js

```diff
@@ -773,6 +775,40 @@ define([
 
         };
 
+        /**
+         * Computes a new set of locations translated from a specified location to a new location for a shape.
+         *
+         * @param {Globe} globe The globe on which to compute a new set of locations.
+         * @param {Location} oldLocation The original reference location.
+         * @param {Location} newLocation The new reference location.
+         * @param {Location[]} locations The locations to translate.
+         *
+         * @return {Location[]} The translated locations.
+         */
+        SurfaceShape.prototype.computeShiftedLocations = function(globe, oldLocation, newLocation, locations) {
+            var newLocations = [];
+            var result = new Vec3(0, 0, 0);
+
+            var oldPoint = globe.computePointFromLocation(oldLocation.latitude, oldLocation.longitude,
+                new Vec3(0, 0, 0));
+            var newPoint = globe.computePointFromLocation(newLocation.latitude, newLocation.longitude,
+                new Vec3(0, 0, 0));
+            var delta = newPoint.subtract(oldPoint);
+
+            for(var i = 0, len = locations.length; i < len; i++)
+            {
+                var point = globe.computePointFromLocation(locations[i].latitude, locations[i].longitude,
+                    result);
+                point = point.add(delta);
+                var newPos = new WorldWind.Position(0, 0, 0);
```

Comment:
`newPos` can be declared before the loop and reused to store the positions returned by `computePositionFromPoint` without creating a new object each time.